gcc/rust/typecheck/rust-hir-type-check-stmt.h

```diff
@@ -42,6 +42,12 @@ class TypeCheckStmt : public TypeCheckBase
   void visit (HIR::ExprStmtWithBlock &stmt) override
   {
     infered = TypeCheckExpr::Resolve (stmt.get_expr (), inside_loop);
+
```

Comment:
I think this check stlll needs to be in the BlockExpr type resolution this is_unit_check_needed is exposed on HIR::Stmts and makes it more general for other cases.